src/Controllers/UserController.js

```diff
@@ -60,11 +60,18 @@ export class UserController extends AdaptableController {
       updateFields._email_verify_token_expires_at = {__op: 'Delete'};
     }
 
-    return this.config.database.update('_User', query, updateFields).then((document) => {
-      if (!document) {
-        throw undefined;
-      }
-      return Promise.resolve(document);
+    var self = this;
+    var checkIfAlreadyVerified = new RestQuery(self.config, Auth.master(self.config), '_User', {username: username, emailVerified: true});
```

Comment:
"when the email field is changed on a _User emailVerified gets set to false" - ic, in that case i think the check here would make sense.  Though the original implementation has `resolve(document)` it now doesn't?

Why would you be in favor of keeping it in both?  I feel like it could be removed in `resend_verification_email` since this already catches it.

Also after this is resolved, please go ahead and make a PR on the original fork so Parse gets the changes as well.